### ISSUE TITLE
Application is corrupting media [lyrics] tags after lookup

### ISSUE THREAD

**Issue by s38b35M5** [NickvisionApps/Tagger#416]
**Description of the issue:**
Changes to media tags are corrupt or otherwise unreadable after applying changes from lookup to lyrics database. Application shows a file system error `Library was changed on disk. Reload` for each file loaded. Occasionally (but not regularly), Tagger reports the corruption and claims to repair it. However, The changes to media files are not read by my media manager `Strawberry Version 1.0.21` until after opening the media and making an arbitrary change to force saving of tag/info data. 

_I can, however view the saved "corrupted" lyrics in_ [picard](https://picard.musicbrainz.org/)

**How can this issue be reproduced?**

1. Load one or more supported file types
2. Select one or more of the media in the list pane and press CTRL-SHIFT-L to query remote lyrics database(s)
3. `Optional` Confirm lyrics data is applied, by clicking the `""Lyrics` option, then dismiss the lyrics window
4. Click `Apply` and observe `Library was changed on disk. Reload` for each file loaded.
5. Click `Reload` or `x` to reload or dismiss the error and exit the application
6. Observe the saved lyrics are not saved in a way media players like Amarok/Clementine/Strawberry can access

**Expected result:**
Saving tags does not result in multiple errors, and saves the tags/lyrics without apparent corruption

**Platform:**
Debian 11 Linux

**Tagger Debug Blob:**
```
org.nickvision.tagger
2023.11.3
GTK 4.12.4
libadwaita 1.4.2
Flatpak
en-US
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

**Attachments:**
[inxi.txt](https://github.com/NickvisionApps/Tagger/files/13620793/inxi.txt) System info
[tagger.zip](https://github.com/NickvisionApps/Tagger/files/13620839/tagger.zip) Before/After media file and screen recording archived in a 7z (inside zip to save space and get around upload filter)

**Comment by nlogozzo** [comment]
> Changes to media tags are corrupt or otherwise unreadable after applying changes from lookup to lyrics database

Does this only happen with looking up Lyrics?

If you have a valid file and just change the title/artist and apply, does that remain ok?
What if you manually edit the Lyrics instead of lookup?

**Comment by s38b35M5** [comment]
> > Changes to media tags are corrupt or otherwise unreadable after applying changes from lookup to lyrics database
> 
> Does this only happen with looking up Lyrics?
> 
> If you have a valid file and just change the title/artist and apply, does that remain ok? What if you manually edit the Lyrics instead of lookup?

I performed some testing today, and can confirm that the following circumstances do NOT result in errors or corrupt files

- Opening a folder with one song and: editing lyrics manually; looking up lyrics with CTRL+SHIFT+L and applying; searching with CTRL+M and applying
- Opening a folder with three songs and: (Same as above)
- Opening an artist folder with two albums of 13 tracks each and: (same as above)

Then I selected an artist with two dozen albums in it, and did a lyric lookup, and the problem with errors seems to have ceased. I will have more time later this week to mess with it, but so far, it seems this problem (which persisted across reboots and different network configs) seems have been transient.

If you have any suggestions to assist troubleshooting, I'm open to them.

Thank you for your time and support.

**Comment by nlogozzo** [comment]
It could have just been a fluke in the flatpak portal setup...

@s38b35M5 has this issue occurred again? or is it gone?

**Comment by s38b35M5** [comment]
> It could have just been a fluke in the flatpak portal setup...
> 
> @s38b35M5 has this issue occurred again? or is it gone?

I just did some extensive testing, and I can't replicate it again. Gremlins...

**Comment by s38b35M5** [comment]
It is happening again, if you would like to take a closer look. May only be with OGG files. Not critical for me, so no rush.

**Comment by nlogozzo** [comment]
Could you send me the specific files that it's happening with?

Email at nlogozzo225@gmail.com

**Comment by s38b35M5** [comment]
> Could you send me the specific files that it's happening with?
> 

Sure thing. I'll share something in the next day or two when I can. Love to help if I can.

**Comment by nlogozzo** [comment]
Yeah having the files is best so we can exam the internal strutture to see if there are any issues...

**Comment by s38b35M5** [comment]
You've got mail 📬

**Comment by nlogozzo** [comment]
@s38b35M5 So I was kind of a dumb dumb and didn't save the email when I cleared my inbox 😅

Could you resend it please?

**Comment by s38b35M5** [comment]
You should have that email now. Thanks for looking into it!
